modyn/models/rho_loss_twin_model/rho_loss_twin_model.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from typing import Any, Optional
+
+import torch
+from modyn.utils import dynamic_module_import
+from torch import nn
+
+logger = logging.getLogger(__name__)
+
+
+class RHOLOSSTwinModel:
+
+    def __init__(self, model_configuration: dict[str, Any], device: str, amp: bool) -> None:
+        self.model = RHOLOSSTwinModelModyn(model_configuration, device, amp)
+        self.model.to(device)
+
+
+class RHOLOSSTwinModelModyn(nn.Module):
+
+    def __init__(self, model_configuration: dict[str, Any], device: str, amp: bool) -> None:
+        super().__init__()
+        self.device = device
+        model_module = dynamic_module_import("modyn.models")
+        rho_model_class = model_configuration["rho_real_model_class"]
+        rho_model_config = model_configuration["rho_real_model_config"]
+        model_handler = getattr(model_module, rho_model_class)
+        # we only need the inner model, not the wrapper
+        self._models = nn.ModuleList(
+            [
+                model_handler(rho_model_config, device, amp).model,
+                model_handler(rho_model_config, device, amp).model,
+            ]
+        )
+        self._models_seen_ids: list[set[int]] = [set(), set()]
+        self._current_model = 0
+
+    def get_extra_state(self) -> Any:
+        return {
+            "models_seen_ids": self._models_seen_ids,
+        }
+
+    def set_extra_state(self, state: dict) -> None:
+        self._models_seen_ids = state["models_seen_ids"]
```

Comment:
yeah, if this is just a single item, not sure why we use a dict

Reply:
see https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.set_extra_state